smartconfig/entry.py

```diff
@@ -0,0 +1,123 @@
+from itertools import chain
+from typing import Any, Dict, NoReturn, Optional, Tuple
+
+from smartconfig import _registry
+from smartconfig._registry import global_configuration, get_attribute, used_paths, unload_defaults
+from smartconfig.exceptions import ConfigurationError, ConfigurationKeyError, InvalidOperation, PathConflict
+
+
+class _ConfigEntryMeta(type):
+    """
+    Metaclass used to define special ConfigEntry behaviors.
+
+    Note: Using this metaclass outside of the library is currently not supported.
+    """
+
+    def __getattribute__(cls, name: str) -> Optional[Any]:
+        """
+        Look up the attribute through the configuration system.
+
+        If the attribute name starts with `_`, normal lookup is done, otherwise _registry.global_configuration is used.
+
+        Args:
+            name: Attribute to lookup
+
+        Raises:
+            ConfigurationKeyError: The attribute doesn't exist.
```

Comment:
I'd say public functions should document exceptions raised by private utility functions, yes.

Reply:
That said, I don't think it is possible to make this function raise `ConfigurationError` unless a bug exists inside the library, since creating the entry will ensure that it has an existing path.